test.gd

```diff
@@ -3,33 +3,189 @@ extends SceneTree
 # To run this script
 # godot -s test.gd
 
-const NUMBER_OF_TEST = 500000
+const NUMBER_OF_TESTS = 500000
+const NUMBER_OF_OBJECTS = 50000  # enough to test and to not run out of memory
 
-const uuid_util = preload('res://uuid.gd')
+var uuid_util = preload('uuid.gd')
 
-func benchmark():
+func benchmark_raw():
   print('Benchmarking ...')
 
-  var begin = OS.get_ticks_msec()
-  var index = 0
+  var uuids := []
+  var begin = Time.get_unix_time_from_system()
 
-  while index < NUMBER_OF_TEST:
+  for i in range(NUMBER_OF_TESTS):
```

Comment:
Better not use `for` loop with large integer.
Godot create a large array to iterate ( https://github.com/godotengine/godot/blob/master/modules/gdscript/gdscript_utility_functions.cpp#L131 ).

If you replace 500000 with 10000000, then the test will crash.

Also, please replace the `i` by `index`. I'm trying to avoid any single letter variables, `index` is more readable than `i` to me.

Reply:
I was wondering about the use of `while` instead of `for`. That explains it. I wasn't aware of that and during testing my system memory monitor did not show any change for that part.

I would use `_` in place of `i` as the variable actually not needed (convention borrowed from python), but Godot does not allow that.

I will revert these changes.